spec/spec_helper_acceptance_litmus.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+require 'lib/dsc_utils_litmus'
+require 'erb'
+require 'serverspec'
+require 'puppet_litmus'
+require 'open3'
+include PuppetLitmus
+
+def run_local_command(command)
+  stdout, stderr, status = Open3.capture3(command)
+  error_message = "Attempted to run\ncommand:'#{command}'\nstdout:#{stdout}\nstderr:#{stderr}"
+  raise error_message unless status.to_i.zero?
+
+  stdout
+end
+
+def single_dsc_resource_manifest(dsc_type, dsc_props)
+  output = "dsc_#{dsc_type} {'#{dsc_type}_test':\n"
+  dsc_props.each do |k, v|
+    if v =~ /^\[.*\]$/
+      output += "  #{k} => #{v},\n"
+    elsif v =~ /^(true|false)$/
+      output += "  #{k} => #{v},\n"
+    elsif v =~ /^{.*}$/
+      output += "  #{k} => #{v},\n"
+    else
+      output += "  #{k} => '#{v}',\n"
+    end
+  end
+  output += "}\n"
+  return output
+end
+
+def escape_value(value)
+  if is_number?(value)
+    value
+  else
+    "'#{value.to_s.gsub("'","''")}\'"
+  end
+end
+
+def is_number?(value)
+  value.respond_to?(:to_i) ? value.to_s == value.to_i.to_s : false
+end
+
+def create_windows_file(folder_path, file_name, content)
+  content = content.gsub(/\"/, "'")
+  manifest = <<-FileManifest
+    file{"#{folder_path}":
+      ensure => directory,
+    }
+
+    file { "#{folder_path}/#{file_name}":
+      content => "#{content}",
+    }
+  FileManifest
+  apply_manifest(manifest)
+end
+
+def puppet_version
+  run_shell('puppet --version', :acceptable_exit_codes => 0).stdout
+end
+
+if ENV['TARGET_HOST'].nil? || ENV['TARGET_HOST'] == 'localhost'
+  puts 'Running tests against this machine !'
+  if Gem.win_platform?
+    set :backend, :cmd
```

Comment:
@tphoney : @RandomNoun7 and I found the problem. On windows localhost it's executing in cmd and not powershell, hence the failure of all the powershell calls :/

Reply:
In further research I'v found that it's true commands run locally are run in cmd, but not because of this setting. This setting belongs to `specinfra`, but we don't end up using it's `run_command` methods. If we used `specinfra`'s we would likely get powershell execution for free because even though the backend is called `cmd`, it actually [builds a powershell script](https://github.com/mizzy/specinfra/blob/master/lib/specinfra/backend/cmd.rb#L13) out of whatever you give to it.

Instead, specinfra is loaded first, but then bolt_spec/run is loaded later which has its own implementation of `run_command` which I think is clobbering `specinfra`'s.

In Bolt land you get a transport choice of `local`, `winrm` or `ssh`. When running commands against a remote node, you get a powershell session because Bolt [made that choice specifically when creating the session](https://github.com/puppetlabs/bolt/blob/master/lib/bolt/transport/winrm/connection.rb#L54). When executing commands on Windows locally they did not make the same choice, instead opting for a [simple call to Open3.capture3](https://github.com/puppetlabs/bolt/blob/master/lib/bolt/transport/local_windows.rb#L71).

We could potentially solve this in Litmus, but I believe the correct place to solve this may be in Bolt. It is actually Bolt that is introducing this inconsistency in behavior. Bolt should be providing a PowerShell wrapper for local command calls on Windows as PowerShell is the default environment most users will expect when running commandline commands on Windows.

Reply:
I just created [BOLT-1429](https://tickets.puppetlabs.com/browse/BOLT-1429) in hopes the Bolt team will agree that this issue is best solved closer to the source, rather than in every individual consumer of `BoltSpec::Run.run_command`.

Reply:
nice catch, it might be worth throwing up a PR to show the bolt team what you want. It looks like they have similar behaviour to what we want for run_task and run_script, were it invokes the powershell helper methods.

Reply:
are we dependant on this change going in ?

Reply:
We are not, it will run as long as we wrap all the calls with "powershell.exe -NoProfile -Nologo -Command ..." which is annoying, but functional. And we have a ticket here FM-8298 to remove that annoyance when Bill's BOLT-1429 gets merged.